### PR TITLE
Undefined xmlns task

### PR DESCRIPTION
**Pull request by benrudolph**
@NoahCarnahan alrightly, let's get this bad boy going on celery

http://manage.dimagi.com/default.asp?224026 🔫 

cc: @proteusvacuum @sravfeyn

### CODE REVIEW

File: corehq/apps/cleanup/tasks.py [hunk 1]
```diff
@@ -1,10 +1,47 @@
...
+from corehq.apps.hqwebapp.tasks import mail_admins_async
 from corehq.apps.cleanup.management.commands.fix_xforms_with_undefined_xmlns import \
     parse_log_message, ERROR_SAVING, SET_XMLNS, MULTI_MATCH, \
     CANT_MATCH, FORM_HAS_UNDEFINED_XMLNS
 
 
+UNDEFINED_XMLNS_LOG_DIR = '/tmp/'
+
+
+def json_handler(obj):
```
Comment (benrudolph) [inline corehq/apps/cleanup/tasks.py]: surprisingly our json_handle doesn't handle making sets serializable: https://github.com/dimagi/dimagi-utils/blob/master/dimagi/utils/web.py#L173. just wrote a new one here, though i doubt there would be any bad consequence to adding it in our own handler...

File: corehq/apps/cleanup/tasks.py [hunk 2]
```diff
@@ -1,10 +1,47 @@
...
+
+from django.core.management import call_command
+
+from corehq.apps.hqwebapp.tasks import mail_admins_async
 from corehq.apps.cleanup.management.commands.fix_xforms_with_undefined_xmlns import \
     parse_log_message, ERROR_SAVING, SET_XMLNS, MULTI_MATCH, \
     CANT_MATCH, FORM_HAS_UNDEFINED_XMLNS
 
 
+UNDEFINED_XMLNS_LOG_DIR = '/tmp/'
```
Comment (NoahCarnahan) [inline corehq/apps/cleanup/tasks.py]: I think we want to store these logs somewhere more permanent than `/tmp/` I don't know if we need to save them indefinitely, but I would think they should stick around a while incase we discover that something has gone wrong. Maybe we should use the same system/policy that we do for the other logs?

Comment (benrudolph) [inline corehq/apps/cleanup/tasks.py]: that's fair, i'll move them to place a little more permanent. fwiw `/tmp` only gets cleared on reboot, which does not happen very often on our machines (or at least shouldn't). though storing it in `/tmp` is like saying it's fine to delete whenever, which is probably more of a concern

Comment (benrudolph) [inline corehq/apps/cleanup/tasks.py]: one thing to note though is that it'll be hard to rotate those logs unless we do significant modifications to the original command. (we rely on our python log handlers to do rotation) so these files will be sitting around in the `log` dir until someone decides to clean them out

Comment (benrudolph) [inline corehq/apps/cleanup/tasks.py]: https://github.com/dimagi/commcare-hq/pull/12367/commits/52e2f848851a61cff9102d4d29cec79fafbef5da

**Comment by benrudolph** [comment]
need to deploy this first though: https://github.com/dimagi/commcarehq-ansible/pull/453

File: corehq/apps/cleanup/tasks.py (hunk 2 above)
Comment (NoahCarnahan) [inline corehq/apps/cleanup/tasks.py]: We could create another task to delete them maybe? (or add it to the end of the current task) Or we can punt until this becomes an issue.

Comment (benrudolph) [inline corehq/apps/cleanup/tasks.py]: i think let's punt til it becomes an issue. they'll be small and easy to delete. worst case we can set up a cron task in ansible (we have many of those) that does this kind of clean up

Comment (NoahCarnahan) [inline corehq/apps/cleanup/tasks.py]: 👍

**Comment by proteusvacuum** [comment]
build fail looks real:
======================================================================

FAIL: Tests the ability to find anomalies that violate our asssumptions

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/mnt/commcare-hq-ro/corehq/apps/cleanup/tests/test_fix_forms_and_apps_with_missing_xmlns.py", line 236, in test_fix_xforms_with_missing_xmlns_task

    self.assertEqual(len(mail.outbox), 1)

AssertionError: 0 != 1

**Comment by benrudolph** [comment]
gah that's probably a difference in the test settings. will look into it thanks

**Comment by proteusvacuum** [comment]
kicked that build

**Comment by benrudolph** [comment]
💃 ✅

**Pull request merged**